pkg/domain/entities/pods.go

```diff
@@ -106,23 +106,143 @@ type PodRmReport struct {
 	Id  string //nolint
 }
 
+// PodCreateOptions provides all possible options for creating a pod and its infra container
+// swagger:model PodCreateOptions
 type PodCreateOptions struct {
-	CGroupParent       string
-	CreateCommand      []string
-	Hostname           string
-	Infra              bool
-	InfraImage         string
-	InfraName          string
-	InfraCommand       string
-	InfraConmonPidFile string
-	Labels             map[string]string
-	Name               string
+	CGroupParent       string            `json:"cgroup-parent,omitempty"`
+	CreateCommand      []string          `json:"create-command,omitempty"`
+	Hostname           string            `json:"hostname,omitempty"`
+	Infra              bool              `json:"infra"`
+	InfraImage         string            `json:"infra-image,omitempty"`
+	InfraName          string            `json:"infra-name,omitempty"`
+	InfraCommand       string            `json:"infra-command,omitempty"`
+	InfraConmonPidFile string            `json:"infra-conmon-pidfile,omitempty"`
+	Labels             map[string]string `json:"labels,omitempty"`
+	Name               string            `json:"name,omitempty"`
+	NetFlags           *NetFlags
 	Net                *NetOptions
```

Comment:
Why net flags and net options those contain redundant information, no?

Reply:
yes and no, the types are different leading the unmarshalling to get confused. This is the sole reason I created the NetFlags and the `GenerateTempNetFlags` function

Reply:
But changing the types of NetOptions would wreak havoc everywhere else in libpod

Reply:
This all stems from the switch from pod specgen to create options. This switch "upwards" in the stack causes the input to change a bit